### PR TITLE
Removed checks for history of operation in the translog flow and updated translog generation settings

### PR DESCRIPTION
**Pull request by saikaranam-amazon**
Signed-off-by: Sai Kumar <karanas@amazon.com>

### Description
Removed checks for history of operation in the translog flow and updated settings
 
### Issues Resolved
N/A
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ankitkala — DISMISSED** [review]

File: src/main/kotlin/org/opensearch/replication/action/changes/TransportGetChangesAction.kt [hunk 1]
```diff
@@ -103,9 +103,7 @@ class TransportGetChangesAction @Inject constructor(threadPool: ThreadPool, clus
                 if(fetchFromTranslog) {
                     try {
                         ops = translogService.getHistoryOfOperations(indexShard, request.fromSeqNo, toSeqNo)
-                    } catch (e: ResourceNotFoundException) {
-                        fetchFromTranslog = false
-                    } catch (e: IllegalStateException) {
+                    } catch (e: Exception) {
                         fetchFromTranslog = false
                     }
```
Comment (krishna-ggk) [inline src/main/kotlin/org/opensearch/replication/action/changes/TransportGetChangesAction.kt:108]: Can we log the exception at DEBUG or TRACE level?

File: src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt [hunk 2]
```diff
@@ -704,7 +706,9 @@ open class IndexReplicationTask(id: Long, type: String, action: String, descript
     private suspend fun setupAndStartRestore(): IndexReplicationState {
         // Enable translog based fetch on the leader(remote) cluster
         val remoteClient = client.getRemoteClusterClient(leaderAlias)
-        val settingsBuilder = Settings.builder().put(INDEX_PLUGINS_REPLICATION_TRANSLOG_RETENTION_LEASE_PRUNING_ENABLED_SETTING.key, true)
+        val settingsBuilder = Settings.builder()
+                .put(INDEX_PLUGINS_REPLICATION_TRANSLOG_RETENTION_LEASE_PRUNING_ENABLED_SETTING.key, true)
+                .put(IndexSettings.INDEX_TRANSLOG_GENERATION_THRESHOLD_SIZE_SETTING.key, ByteSizeValue(32, ByteSizeUnit.MB))
         val updateSettingsRequest = remoteClient.admin().indices().prepareUpdateSettings().setSettings(settingsBuilder).setIndices(leaderIndex.name).request()
```
Comment (krishna-ggk) [inline src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt:712]: The translog gen size is a significant change. Can you elaborate that in the commit message and PR?

Also can you include a comment with reasoning/context for the same?

**Review by krishna-ggk — DISMISSED** [review]
LGTM, just minor comments

File: src/main/kotlin/org/opensearch/replication/action/changes/TransportGetChangesAction.kt (hunk 1 above)
Comment (saikaranam-amazon) [reply src/main/kotlin/org/opensearch/replication/action/changes/TransportGetChangesAction.kt]: Done

File: src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt (hunk 2 above)
Comment (saikaranam-amazon) [reply src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt:719]: Done - updated

**Review by krishna-ggk — APPROVED** [review]

**Review by tbhanu-amzn — APPROVED** [review]

**Pull request merged**